### PR TITLE
Branch update exercise history

### PR DESCRIPTION
**Pull request by L-K-Chng**
Havent actually done up the update exercise history thing yet but added on more junit tests and assertions.

### CODE REVIEW

File: src/main/java/seedu/duke/exercisegenerator/GenerateExercise.java [hunk 1]
```diff
@@ -78,9 +80,17 @@ public ArrayList<ExerciseData> generateFilteredDifficultySetFrom(ArrayList<Exerc
         return filteredExerciseList;
     }
 
-    //lipkuang - Added new filter for workout type.
+    /**
+     * Returns an exercise list which is filtered according to the workout type
```
Comment (ghzr0) [inline src/main/java/seedu/duke/exercisegenerator/GenerateExercise.java:84]: Good comment for better understanding of what function does

File: src/test/java/seedu/duke/exercisegenerator/GenerateWorkoutTypeExerciseTest.java [hunk 2]
```diff
@@ -10,90 +10,144 @@
 import static org.junit.jupiter.api.Assertions.assertNotEquals;
 
 public class GenerateWorkoutTypeExerciseTest {
+    private static final String UPPER = "upper";
```
Comment (ghzr0) [inline src/test/java/seedu/duke/exercisegenerator/GenerateWorkoutTypeExerciseTest.java:13]: Good usage of constants for frequently used terms

**Review by ghzr0 — COMMENTED** [review]
Good addition to code standards

File: src/test/java/seedu/duke/exercisegenerator/GenerateBodyExerciseTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+public class GenerateBodyExerciseTest {
+    private static String BODY = "body only";
+
+    /**
+     * Checks if program accurately filters body weight exercises from
+     * entire list of data
+     */
```
Comment (EangJS) [inline src/test/java/seedu/duke/exercisegenerator/GenerateBodyExerciseTest.java:17]: Can explain the test more i.e to say like explain the expected output

**Review by EangJS — APPROVED** [review]
Good use of static final strings to replace constants
Good addition of JUnitTests to improve code testing coverage

File: src/test/java/seedu/duke/exercisegenerator/GenerateBodyExerciseTest.java (hunk 3 above)
Comment (L-K-Chng) [reply src/test/java/seedu/duke/exercisegenerator/GenerateBodyExerciseTest.java:17]: ok, will update in next pr

**Pull request merged**